models/nft/ethereum/nft_ethereum_loans.sql

```diff
@@ -0,0 +1,379 @@
+),
+
+arcade_v2_vault_withdrawn_nfts as (
+    select e.`from` as vault, p.to as borrower, p.tokenId, p.token as contract_address, p.call_block_time, e.evt_tx_hash
+    from {{ source('pawnfi_v201_ethereum','AssetVault_call_withdrawERC721') }} p inner join {{ source('erc721_ethereum','evt_Transfer') }}  e on p.call_block_time=e.evt_block_time and p.call_tx_hash=e.evt_tx_hash
+),
+
+arcade_v2_vault_total_nfts as (
+    select * 
+    from (
+        select d.vault, d.tokenId, d.contract_address as collectionContract, d.call_block_time as d_block_time, w.call_block_time as w_block_time, 
+        row_number() over (partition by d.vault, d.tokenId, d.contract_address order by w.call_block_time) as r
+        from arcade_v2_vault_deposited_nfts d left join arcade_v2_vault_withdrawn_nfts w on d.call_block_time<w.call_block_time and d.vault=w.vault and d.tokenId=w.tokenId and d.contract_address=w.contract_address
+    ) t
+    where r is null or r = 1
+),
+
+arcade_v2_base as (
+    select *, 365 * 1.0 / duration * (interest_rate_raw / 1e18 / 100) as apr
+    from (
+        select call_tx_hash as evt_tx_hash, call_block_time as evt_block_time, borrower, lender, contract_address, output_loanId as loanId, 'v2' as version,
+            get_json_object(terms, '$.collateralId') as collateralId,
+            get_json_object(terms, '$.collateralAddress') as collateralAddress,
+            get_json_object(terms, '$.payableCurrency') as currency,
+            get_json_object(terms, '$.principal') as principal_raw,
+            get_json_object(terms, '$.interestRate') as interest_rate_raw,
+            get_json_object(terms, '$.durationSecs') / 86400 as duration
+        from {{ source('pawnfi_v201_ethereum','LoanCore_call_startLoan') }}
+        where call_success=true
+        union all
+        select call_tx_hash as evt_tx_hash, call_block_time as evt_block_time, borrower, lender, contract_address, output_newLoanId as loanId, 'v2-r' as version,
+            get_json_object(terms, '$.collateralId') as collateralId,
+            get_json_object(terms, '$.collateralAddress') as collateralAddress,
+            get_json_object(terms, '$.payableCurrency') as currency,
+            get_json_object(terms, '$.principal') as principal_raw,
+            get_json_object(terms, '$.interestRate') as interest_rate_raw,
+            get_json_object(terms, '$.durationSecs') / 86400 as duration
+        from {{ source('pawnfi_v201_ethereum','LoanCore_call_rollover') }}
+        where call_success=true
+    ) t
+),
+
+arcade_v2 as (
+    select l.*, case when (r.evt_block_time is null and l.evt_block_time + interval '1 day' * duration < current_date) then l.evt_block_time else null end as repay_time
+    from arcade_v2_base l
+    left join {{ source('pawnfi_v201_ethereum',' LoanCore_evt_LoanClaimed') }} r

```

Comment:
Hi @ivankitanovski, please add/define this source(and other source tables) in sources.yml. This may help to pass the tests.